base/sort.jl

```diff
@@ -82,16 +82,80 @@ issorted(itr;
     lt=isless, by=identity, rev::Bool=false, order::Ordering=Forward) =
     issorted(itr, ord(lt,by,rev,order))
 
-function select!(v::AbstractVector, k::Union{Int,OrdinalRange}, o::Ordering)
+function partialsort!(v::AbstractVector, k::Union{Int,OrdinalRange}, o::Ordering)
     inds = indices(v, 1)
     sort!(v, first(inds), last(inds), PartialQuickSort(k), o)
-    v[k]
+
+    if k isa Integer
```

Comment:
In general I'd agree with you, but in the present case that would force duplicating the two lines above, that's why I chose the `if` approach.

BTW, there seems to be a lack in the API: with `getindex`, you can either get an array or a scalar, but with `view`/`@view` you always get a subarray. Shouldn't there be a way to get the same behavior as `getindex`? Then one wouldn't have to check types manually like this (which could be hard or impossible in more complex cases where the index can be of any type.)